scripts/databricks_install.py

```diff
@@ -0,0 +1,255 @@
+#!/usr/bin/env python
+
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+# This script installs Recommenders/reco_utils as a library into Databricks
+
+import argparse
+import textwrap
+import os
+
+import shutil
+import sys
+from urllib.request import urlretrieve
+
+## requires databricks-cli to be installed:
+## and requires authentication to be configured
+from databricks_cli.configure.provider import ProfileConfigProvider
+from databricks_cli.configure.config import _get_api_client
+from databricks_cli.clusters.api import ClusterApi
+from databricks_cli.dbfs.api import DbfsApi
+from databricks_cli.libraries.api import LibrariesApi
+
+from requests.exceptions import HTTPError
+
+CLUSTER_NOT_FOUND_MSG = """
+    Cannot find the target cluster {}. Please check if you entered the valid id. 
+    Cluster id can be found by running 'databricks clusters list', which returns a table formatted as:
+
+    <CLUSTER_ID>\t<CLUSTER_NAME>\t<STATUS>
+    """
+
+CLUSTER_NOT_RUNNING_MSG = """
+    Cluster {0} found, but it is not running. Status={1}
+    You can start the cluster with 'databricks clusters start --cluster-id {0}'.
+    Then, check the cluster status by using 'databricks clusters list' and
+    re-try installation once the status becomes 'RUNNING'.
+    """
+""
+
+## Variables for operationalization:
+COSMOSDB_JAR_FILE_OPTIONS = {
+  '3': 'https://search.maven.org/remotecontent?filepath=com/microsoft/azure/azure-cosmosdb-spark_2.2.0_2.11/1.1.1/azure-cosmosdb-spark_2.2.0_2.11-1.1.1-uber.jar',
+  '4': 'https://search.maven.org/remotecontent?filepath=com/microsoft/azure/azure-cosmosdb-spark_2.3.0_2.11/1.2.2/azure-cosmosdb-spark_2.3.0_2.11-1.2.2-uber.jar',
+  '5': 'https://search.maven.org/remotecontent?filepath=com/microsoft/azure/azure-cosmosdb-spark_2.4.0_2.11/1.3.5/azure-cosmosdb-spark_2.4.0_2.11-1.3.5-uber.jar'
+}
+
+PYPI_O16N_LIBS=["azure-cli==2.0.56", "azureml-sdk[databricks]==1.0.8", "pydocumentdb==2.3.3"]
+
+
+def create_egg(path_to_recommenders_repo_root=os.getcwd(),local_eggname='Recommenders.egg', overwrite=False):
+  """
+  Packages files in the reco_utils directory as a .egg file that can be uploaded to dbfs and installed as a library on a databricks cluster.
+  """
+  ## create the zip archive:
+  myzipfile = shutil.make_archive('reco_utils',
+                'zip', 
+                root_dir = path_to_recommenders_repo_root, 
+                base_dir = 'reco_utils'
+              )
+
+  ## make sure extension is .egg
+  if local_eggname[-4:] != ".egg":
+    print('appending .egg to end of name.')
+    logal_eggname = local_eggname+'.egg'
+  
+  ## overwrite egg if it previously existed
+  if os.path.exists(local_eggname) and overwrite:
+    os.unlink(local_eggname)
+  os.rename(myzipfile,local_eggname)
+  return local_eggname
+
+def prepare_for_operationalization(cluster_id, api_client, status=None, dbfs_path="dbfs:/FileStore/jars", overwrite=False):
```

Comment:
do you need default values here for dbfs_path and overwrite? it's one more thing to keep consistent with argparse defaults.

Reply:
also, do you need status? basically we just want the spark_version? I would suggest just putting the spark_version as an input and extracting that from the status external to this function